flow/alerting/classifier.go

```diff
@@ -249,10 +249,38 @@ func GetErrorClass(ctx context.Context, err error) (ErrorClass, ErrorInfo) {
 
 	var myErr *mysql.MyError
 	if errors.As(err, &myErr) {
-		return ErrorOther, ErrorInfo{
+		// https://mariadb.com/kb/en/mariadb-error-code-reference
+		myErrorInfo := ErrorInfo{
 			Source: ErrorSourceMySQL,
 			Code:   strconv.Itoa(int(myErr.Code)),
 		}
+		switch myErr.Code {
+		case 1037, 1038, 1041, 3015:
+			return ErrorNotifyOOM, myErrorInfo
+		case 1040: // ER_CON_COUNT_ERROR
+			return ErrorNotifyConnectivity, myErrorInfo
+		case 1021, // ER_DISK_FULL
+			1044, // ER_DBACCESS_DENIED_ERROR
+			1045, // ER_ACCESS_DENIED_ERROR
+			1049, // ER_BAD_DB_ERROR
+			1051, // ER_BAD_TABLE_ERROR
+			1053, // ER_SERVER_SHUTDOWN
+			1102, // ER_WRONG_DB_NAME
+			1103, // ER_WRONG_TABLE_NAME
+			1109, // ER_UNKNOWN_TABLE
+			1119, // ER_STACK_OVERRUN
+			1129, // ER_HOST_IS_BLOCKED
+			1130, // ER_HOST_NOT_PRIVILEGED
+			1133, // ER_PASSWORD_NO_MATCH
+			1135, // ER_CANT_CREATE_THREAD
+			1152, // ER_ABORTING_CONNECTION
+			1194, // ER_CRASHED_ON_USAGE
+			1195, // ER_CRASHED_ON_REPAIR
```

Comment:
some of these seem like core database issues rather than connectivity ones, can either make a new class or map to other?

Reply:
yes, in meantime I bucketed them on this error type